src/logius/fix-md-elements.js

```diff
@@ -0,0 +1,84 @@
+// @ts-check
+// Module logius/fix-table
+// add table class simple to all tables
+
+export const name = "logius/fix-md-elements";
+
+export function run(conf) {
+  addClassTables(conf);
+  addFigureImg(conf);
+  addClassCode(conf);
+}
+// todo check if algorithm is correct!
+function addClassTables(conf) {
+  // todo do nothing if this config is not set?
+  if (!conf.nl_markdownTableClass) {
+    conf.nl_markdownTableClass = "simple";
+  }
+  [...document.querySelectorAll("[data-format=markdown]:not(body)")].forEach(
+    section =>
+      section.querySelectorAll("table").forEach(table => {
+        // table.classList.add("complex");
+        table.classList.add(conf.nl_markdownTableClass);
+      })
+  );
+}
+
+// todo check non happy flows
+function addFigureImg(conf) {
+  if (
+    !conf.nl_markdownEmbedImageInFigure ||
+    conf.nl_markdownEmbedImageInFigure == false
+  ) {
+    return;
+  }
+
+  [...document.querySelectorAll("[data-format=markdown]:not(body)")].forEach(
+    section =>
+      section.querySelectorAll("img").forEach(img => {
+        // filter out images that already are embedded in a figure element
+        if (img.parentNode.nodeName != "FIGURE") {
+          const figure = document.createElement("figure");
+          const figcaption = document.createElement("figcaption");
+          // todo
+          const filePath = img.getAttribute("src");
+          const extractFilename = path => {
+            let pathArray = path.split("/");
+            const lastIndex = pathArray.length - 1;
+            const filename = pathArray[lastIndex];
+            pathArray = filename.split(".");
+            return pathArray[0];
+          };
+          const id = extractFilename(filePath);
+          figure.setAttribute("id", id);
+          const caption = img.getAttribute("alt")
+            ? `${img.getAttribute("alt")}`
+            : "todo_caption";
+          figcaption.innerText = caption;
+          const cloneImg = img.cloneNode(false);
+          if (!img.getAttribute("title")) {
+            cloneImg.setAttribute("title", caption);
+          }
+          figure.appendChild(cloneImg);
+          figure.appendChild(figcaption);
+          img.parentNode.insertBefore(figure, img);
+          img.remove();
+        }
+      })
+  );
```

Comment:
I think you want something like the following. You will need to import `addId` and `showError` from utils, and hyperHTML as well (this is untested... it's just a suggestion): 

```suggestion
  [...document.querySelectorAll("section[data-format=markdown] img")]
  .filter(img => !img.closest("figure"))
  .forEach(img => {
    if (!img.alt) {
      const msg = " ... warning message about missing alt ... ";
      const hint = " ... helpful text here.... ";
      showError(msg, name, { elements: [img], hint });
    }
    const alt = img.alt;
    const figure = html`<figure>
      ${img.clone(true)}
      <figcaption>${alt}</figcaption>
    </figure>`;
    addId(figure);
    img.replaceWith(figure);
  });
```